pkg/cache/cache_test.go

```diff
@@ -0,0 +1,214 @@
+package cache_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	"github.com/openshift-psap/special-resource-operator/pkg/cache"
+	"github.com/openshift-psap/special-resource-operator/pkg/clients"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestCache(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Cache Suite")
+}
+
+func resetGlobals() {
+	clients.Interface = nil
+
+	cache.Node.Count = 0xDEADBEEF
+	cache.Node.List = &unstructured.UnstructuredList{
+		Object: map[string]interface{}{},
+		Items:  []unstructured.Unstructured{},
+	}
+}
+
+var _ = Describe("Cache", func() {
+	Context("Nodes", func() {
+		var (
+			ctrl        *gomock.Controller
+			mockClients *clients.MockClientsInterface
+		)
+
+		BeforeEach(resetGlobals)
+
+		BeforeEach(func() {
+			ctrl = gomock.NewController(GinkgoT())
+			mockClients = clients.NewMockClientsInterface(ctrl)
+
+			clients.Interface = mockClients
+		})
+
+		AfterEach(func() {
+			ctrl.Finish()
+		})
+		AfterEach(resetGlobals)
+
+		listMatcher := gomock.AssignableToTypeOf(&unstructured.UnstructuredList{})
+
+		When("machingLabels=nil", func() {
+			When("force=false", func() {
+				When("no taints", func() {
```

Comment:
I must say I really don't like those `When`, I don't think they add too much value and the code is less clear IMO.
What benefits to we get from them? why not:
* Describe for the package level
* Context for the function level
* It for each on of the cases we want to test inside that function we are testing